src/Passes/Source/RuleTransformationPass/Configuration.cpp

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Commandline/ConfigurationManager.hpp"
+#include "RuleTransformationPass/Configuration.hpp"
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    void RuleTransformationPassConfiguration::setup(ConfigurationManager& config)
+    {
+        config.setSectionName("Pass configuration", "Configuration of the pass and its corresponding optimisations.");
+        config.addParameter(delete_dead_code_, "delete-dead-code", "Deleted dead code.");
+        config.addParameter(clone_functions_, "clone-functions", "Clone functions to ensure correct qubit allocation.");
+        config.addParameter(
+            transform_execution_path_only_, "transform-execution-path-only", "Transform execution paths only.");
+
+        config.addParameter(
+            max_recursion_, "max-recursion", "Defines the maximum recursion when unrolling the execution path");
+
+        config.addParameter(
+            assume_no_exceptions_, "assume-no-except", "Assumes that no exception will occur during runtime.");
+
+        config.addParameter(reuse_qubits_, "reuse-qubits", "Use to define whether or not to reuse qubits.");
+        config.addParameter(annotate_qubit_use_, "annotate-qubit-use", "Annotate the number of qubits used");
+
+        config.addParameter(reuse_results_, "reuse-results", "Use to define whether or not to reuse results.");
+        config.addParameter(annotate_result_use_, "annotate-result-use", "Annotate the number of results used");
+
+        config.addParameter(
+            entry_point_attr_, "entry-point-attr", "Specifies the attribute indicating the entry point.");
+
+        config.addParameter(
+            simplify_prior_transformation_, "simplify-prior-transform",
+            "When active, the IR is simplified using LLVM passes before transformation.");
+
+        // Not implemented yet
+        config.addParameter(group_measurements_, "group-measurements", "NOT IMPLEMENTED - group-measurements");
+        config.addParameter(one_shot_measurement_, "one-shot-measurement", "NOT IMPLEMENTED - one-shot-measurement");
+    }
```

Comment:
What do you think about doing this in the constructor of `RuleTransformationPassConfiguration`?

Reply:
The configuration may be created ahead of time (I.e. before the ConfigurationManager exists). Therefore, it is not possible to do this via the constructor.